trace/metric_test.go

```diff
@@ -0,0 +1,157 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	dto "github.com/prometheus/client_model/go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCounter_Add(t *testing.T) {
+	type fields struct {
+		counter prometheus.Counter
+	}
+	type args struct {
+		count float64
+	}
+	test := struct {
```

Comment:
We are trying to move to map based table-tests and also parallelizing them.
Check this out: https://github.com/beatlabs/patron/blob/b785f11f89ffd990fd0ac67d2be44e0f1f62fc9f/client/kafka/v2/kafka_test.go#L12

Reply:
made use of table-tests, parallelise execution of the tests.